pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -260,14 +259,17 @@ func translateRoute(in *networking.HTTPRoute,
 	proxyLabels model.LabelsCollection,
 	gatewayNames map[string]bool) (*route.Route, error) {
 
+	// When building routes, its okay if the target cluster cannot be
+	// resolved Traffic to such clusters will blackhole.
+
 	// Match by source labels/gateway names inside the match condition
 	if !sourceMatchHTTP(match, proxyLabels, gatewayNames) {
-		return nil, fmt.Errorf("no source match: %v", match)
+		return nil, nil
 	}
 
 	// Match by the destination port specified in the match condition
 	if match != nil && match.Port != 0 && match.Port != uint32(port) {
```

Comment:
Can't this if and the one above be removed now? They were there to present helpful errors and now are just `nil,nil`

Reply:
we still need to return nil if the match fails, right?